### PR TITLE
contact page

### LINKED ISSUE #14: Contact page

### PR DESCRIPTION
**Pull request by parth4apple**
### Administrative Info
[GitHub Issue #14]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
I implemented the contact page and its styling. The page is responsive to different screen sizes.

### Testing
How did you confirm your changes worked? 
The page has the same behavior as the figma model.

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
Screen Shot 2022-01-25 at 10 36 22 AM
Screen Shot 2022-01-25 at 10 36 40 AM

### CODE REVIEW

**Review by dnanjunda — CHANGES_REQUESTED** [review]
1. This isn't super important, but can you make the background color for the dropdown gray instead of white, like in the designs?
2. Add a bit more padding for the "continue" button. It should be similar to the "send" button on the other screen.

Your code:
continue
Figma:
continue_figma

3. For most of the mobile screens, the text is way too big. Also, try to reduce the height of the component with the message box.
contact_mobile


4. Everything seems a bit too big to me. Make sure all the components/text you use are responsive.
contact


5. Merge main with your branch so that you have the navbar for this page.

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Looks better Parth! There are a few more changes needed:

1. Kind of a nitpick but make the send and continue buttons less rounded and the same width like in the figma:

Your code:
button

Figma:
button_figma

2. Add form validation to check that a team is selected (in the first component itself) and that the message box isn't empty (when they click send). For both cases, display red text above the component showing the error. So, if a team isn't selected, display "Please select a team to send the message to." If the message is empty, you can say "Message body cannot be blank."
3. For the thank you popup, increase the font size a bit and center it within the box as well.
4. For a specific range of screen width (750 to 900 px), the contact component overlaps with the navbar. You can either add a media query to reduce the width of the component for this range or make the width of the navbar be responsive. If possible, try the latter since the navbar is taking up a chunk of the screen anyway.
contact

5. For some of the smaller screen sizes, the dropdown text overflows. Same with the text in the thank you popup:
mobile_contact

6. Increase the width of the thank you popup on mobile screens to almost the same as the component. This looks a bit weird:
contact_popup

File: client/src/css/contactStyles.css [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  font-size: min(20px, 2vw);
+  color: #53575a;
+}
+.continueButton {
+  position: absolute;
+  width: min(15vw, 191px);
+  height: 10%;
+  right: 20px;
+  bottom: 20px;
+  text-align: center;
```
Comment (petabite) [inline client/src/css/contactStyles.css:59]:
```suggestion
  text-align: center;
  cursor: pointer;
```
UX improvement: This should make it more obvious that the "Continue" button is clickable.

File: client/src/css/contactStyles.css [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  border: 0.5px solid #9b9a9a;
+  box-sizing: border-box;
+  border-radius: 15px;
+}
+
+.sendButton {
+  background: #8dc442;
+  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+  border-radius: 20px;
+  border: none;
```
Comment (petabite) [inline client/src/css/contactStyles.css:154]:
```suggestion
  border: none;
  cursor: pointer;
```
same for "Send" button

File: client/src/css/contactStyles.css [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+  color: #ffffff;
+  display: flex;
+  justify-content: space-between;
+}
+
+.cancel {
+  padding-right: 10px;
+  background: none;
+  border: none;
```
Comment (petabite) [inline client/src/css/contactStyles.css:115]:
```suggestion
  border: none;
  cursor: pointer;
```
same for "X" button

File: client/src/css/contactStyles.css [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  text-align: center;
+  font-size: min(18px, 1.5vw);
+}
+
+.smallCancel {
+  border: none;
+  background: none;
+  position: absolute;
+  top: 5%;
+  left: 2%;
```
Comment (petabite) [inline client/src/css/contactStyles.css:197]:
```suggestion
  left: 2%;
  cursor: pointer;
```
small "x" on the "Thank you" modal

File: client/src/css/contactStyles.css [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+  }
+  .sendButton,
+  .continueButton {
+    width: min(100px, 50vw);
+    font-size: min(20px, 5vw);
+  }
+  .selection,
+  .selection2,
+  .inputArea {
+    font-size: min(10px, 4vw);
```
Comment (petabite) [inline client/src/css/contactStyles.css]: On mobile views, the font is extremely small:

image

It looked better when I did this though:
```suggestion
    font-size: max(10px, 4vw);
```
(but you may have a better solution)

**Review by petabite — CHANGES_REQUESTED** [review]
Nice! Responsiveness is looking good on my end.

Some things I noticed:
- Some cursors (see comments)
- Font size on mobile view (see comment)
- The form validation only works the first time. If I send the message the first time successfully and click the "X" in the top right corner and try to send an empty message, it lets me send.

But other than that, it's looking good!

File: package.json [hunk 6]
```diff
@@ -17,7 +17,8 @@
     "dotenv": "^11.0.0",
     "express": "^4.17.2",
     "express-validator": "^6.14.0",
-    "mongoose": "^6.1.6"
+    "mongoose": "^6.1.6",
+    "styled-components": "^5.3.3"
```
Comment (petabite) [inline package.json]:
```suggestion
```
Oops, looks like you added `styled-components` to the backend package.json again.

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Besides the things Philip noted above, here's what I noticed:

1. For the error text, I meant put it above the whole component not just the message box. The placement looks a bit weird right now.
2. Also, the error for selecting a team should be within the first component itself (the one that asks you to choose a team). Don't let the user continue without selecting a team.
3. Not sure what you changed, but the font size is too small now on mobile screens. This is mainly the text in the message box, the dropdown, and the buttons. 
4. For the thank you popup, make it wider for ipad/tablet screens (750 to 1000 px) like you did for mobile. It looks squished right now:
thankyou_popup

**Review by petabite — APPROVED** [review]
👌

**Review by dnanjunda — APPROVED** [review]
LGTM! Great work Parth

**Pull request merged**